ci/scripts/install_dask.sh

```diff
@@ -35,4 +35,4 @@ else
 fi
 
 # additional dependencies needed for dask's s3 tests
-pip install moto[server] flask requests
+pip install "moto[server]<5" flask requests
```

Comment:
Should we add a reference to the original issue?
```suggestion
# Moto 5 results in timeouts in s3 tests: https://github.com/dask/dask/issues/10869
pip install "moto[server]<5" flask requests
```

Reply:
Yes, good idea, thanks!